src/libraries/Microsoft.PowerFx.Interpreter/Functions/SetFunction.cs

```diff
@@ -62,29 +64,31 @@ protected override bool CheckTypes(TexlNode[] args, DType[] argTypes, IErrorCont
             Contracts.AssertValue(errors);
             Contracts.Assert(MinArity <= args.Length && args.Length <= MaxArity);
 
-            nodeToCoercedTypeMap = null;
-            returnType = DType.Boolean;
-
             var arg0 = argTypes[0];
 
             var firstName = args[0].AsFirstName();
-
-            // Global-scoped variable name should be a firstName.
-            if (firstName == null)
-            {
-                errors.EnsureError(DocumentErrorSeverity.Severe, args[0], TexlStrings.ErrNeedValidVariableName_Arg, Name, args[0]);
-                return false;
-            }
-
-            var arg1 = argTypes[1];
-
-            if (!arg0.Accepts(arg1))
-            {
-                errors.EnsureError(DocumentErrorSeverity.Critical, args[1], ErrBadType);
-                return false;
-            }
-
-            return true;
+
+            if (firstName != null)

```

Comment:
> if (firstName != null) [](http://example.com/codeflow?start=11&length=23)

This is enforcing more checks at bind time!

If the Bind isn't successful (mutable symbol), then the IR won't even have a slot anymore to attempt to set.

Reply:
Nice improvement